play.pokemonshowdown.com/src/battle-team-editor.tsx

```diff
@@ -863,11 +873,30 @@ class TeamTextbox extends preact.Component<{ editor: TeamEditorState, onChange?:
 
 		const pokepaste = /^https?:\/\/pokepast.es\/([a-z0-9]+)(?:\/.*)?$/.exec(value)?.[1];
 		if (pokepaste) {
+			this.editor.fetching = true;
 			Net(`https://pokepast.es/${pokepaste}/json`).get().then(json => {
 				const paste = JSON.parse(json);
-				// make sure it's still there:
-				const valueIndex = this.textbox.value.indexOf(value);
-				this.replace(paste.paste.replace(/\r\n/g, '\n'), valueIndex, valueIndex + value.length);
+				const pasteTxt = paste.paste.replace(/\r\n/g, '\n');
+				if (this.textbox) {
+					// make sure it's still there:
+					const valueIndex = this.textbox.value.indexOf(value);
+					this.replace(paste.paste.replace(/\r\n/g, '\n'), valueIndex, valueIndex + value.length);
+				} else {
+					this.editor.import(pasteTxt);
+				}
+				const notes = paste["notes"] as string;
+				if (notes.startsWith("Format: ")) {
+					const formatid = toID(notes.slice(8));
+					if (BattleFormats[formatid]) {
+						this.editor.setFormat(formatid);
+					}
```

Comment:
```suggestion
					this.editor.setFormat(formatid);
```
Probably we should set this for past formats, too.